herd/AArch64Sem.ml

```diff
@@ -245,6 +245,52 @@ module Make
 
       and str sz rs rd kr ii = do_str sz AArch64.N rs (get_ea rd kr ii) ii
 
+      (* Store Pair *)
```

Comment:
I have a double, should not the address increment be 4 and 8 for V32 and V64,  respectively?

Reply:
I think we are talking about the same thing. but as bytes, versus bits, ie `sizeof(byte) * tr_variant(V64) = 8 * MachSize.Quad = 8 * 8 = 64` and `sizeof(byte) * tr_variant(V32) = 8 * MachSize.Word = 8 * 4 = 32`

I think we could simplify this like you say, perhaps as one single function `incr_a` which takes the variant as an argument and can be inlined as a monadic op such as `M.unitT a >>| (M.address_of a >>= incr_a var >>= M.deref)` in stp

Reply:
Hum, I was not clear I guess, this how i see things:

I thought the address "granule" is always the byte. Hence +1 means addressing next byte. +4 then next int32_t etc.

Hence, with concrete value of location x being say, 0x100, with X1=x initially
LDP W0,W1,[X2]
will load the contents of location 0x100 into X0 and the contents of location 0x104 into X1.

Hence, the increment to effective address performed by ``ldp`` would be 4.

Reply:
I see what you mean now, yeah I will fix that